mongo/client.go

```diff
@@ -471,30 +471,48 @@ func (c *Client) endSessions(ctx context.Context) {
 }
 
 func (c *Client) configureAutoEncryption(args *options.ClientOptions) error {
-	c.encryptedFieldsMap = args.AutoEncryptionOptions.EncryptedFieldsMap
+	aeOpts := args.AutoEncryptionOptions
+	c.encryptedFieldsMap = aeOpts.EncryptedFieldsMap
 	if err := c.configureKeyVaultClientFLE(args); err != nil {
 		return err
 	}
 
-	if err := c.configureMetadataClientFLE(args); err != nil {
-		return err
-	}
-
-	mc, err := c.newMongoCrypt(args.AutoEncryptionOptions)
+	mc, err := c.newMongoCrypt(aeOpts)
 	if err != nil {
 		return err
 	}
 
 	// If the crypt_shared library was not loaded, try to spawn and connect to mongocryptd.
 	if mc.CryptSharedLibVersionString() == "" {
-		mongocryptdFLE, err := newMongocryptdClient(args.AutoEncryptionOptions)
+		c.mongocryptdFLE, err = newMongocryptdClient(aeOpts)
 		if err != nil {
 			return err
 		}
-		c.mongocryptdFLE = mongocryptdFLE
 	}
 
-	c.configureCryptFLE(mc, args.AutoEncryptionOptions)
+	kr := keyRetriever{coll: c.keyVaultCollFLE}
+	var cir collInfoRetriever
+	bypass := aeOpts.BypassAutoEncryption != nil && *aeOpts.BypassAutoEncryption
+	if !bypass {
+		if args.MaxPoolSize != nil && *args.MaxPoolSize == 0 {
+			c.metadataClientFLE = c
+		} else {
+			c.metadataClientFLE, err = c.getOrCreateInternalClient(args)
+			if err != nil {
+				return err
+			}
+		}
+		cir.client = c.metadataClientFLE
+	}
+
+	c.cryptFLE = driver.NewCrypt(&driver.CryptOptions{
+		MongoCrypt:           mc,
+		CollInfoFn:           cir.cryptCollInfo,
+		KeyFn:                kr.cryptKeys,
+		MarkFn:               c.mongocryptdFLE.markCommand,
+		TLSConfig:            aeOpts.TLSConfig,
+		BypassAutoEncryption: bypass,
+	})
```

Comment:
This refactor seems unrelated to DEK cache lifetime. I think this should be reverted to avoid confusing the scope of this PR.